### DISCUSSION TITLE
can't update in mongodb

### DISCUSSION THREAD

**Discussion by kpouters** [mikro-orm/mikro-orm#4048] [Q&A]
**Describe the bug**

<img width="366" alt="image" src="https://user-images.githubusercontent.com/113222928/218983103-c4d408fd-510c-4032-bfb6-3998e957d640.png">

I use MongoDB, and I can create and retrieve data, but in some situations, it doesn't update.

1. Update via assign

<img width="992" alt="image" src="https://user-images.githubusercontent.com/113222928/218982458-eb5f2278-f87d-4d8b-a9db-50a1c81ef8a5.png">

2. multiple levels deep update

ex) 

a.b.c = "test" ;
em.flush();

**Comment by B4nan** [comment]
Drop the `merge: true`, that is not doing what you want, you will never get any update queries as you literally tell the orm to merge the state - this option controls `em.merge()`. I'd say drop all the options, you clearly just try things without knowing what they do and what are their defaults (both `mergeObjects` and `updateNestedEntities` default to true).

Screenshots won't help much, provide a (complete) reproduction if you need help. 

If you want to update nested entity graph, be sure to read the docs carefully, you _need_ to load the entities yourself before you assign their FKs if you want to update anything. Don't expect the assign to check anything from the database for you, it is a sync method.

https://mikro-orm.io/docs/entity-helper#updating-deep-entity-graph
